### ISSUE TITLE
Inconsistent `UserDict.__contains__` behavior (vs `collections.abc.Mapping.__contains__`)

### ISSUE THREAD

**Issue by Conchylicultor** [python/cpython#91784]
**tl;dr;**

`Mapping.__contains__` calls `__getitem__`, while `UserDict.__contains__` does not.

**Description**

`collections.UserDict` is great when you want an easy custom `dict`.

* `dict` is very tricky to subclass (to properly handle `.update`, `.__init__`,...)
* `collections.abc.Mapping` is very boilerplate (require implementing `__len__`, `__iter__`,...)

For example, `UserDict` (should) make it trivial to create a dict with lowercase keys:

```python
class LowercaseDict(collections.UserDict):

  def __getitem__(self, key):
    return super().__getitem__(key.lower())

  def __setitem__(self, key, value):
    super().__setitem__(key.lower(), value)


d = LowercaseDict(MY_KEY=0)  # __init__ correctly normalised
d.update({'OTHER_KEY': 1})  # update correctly normalised
d['Hello'] = d['other_KEY']
print(d)  # All keys normalized {'my_key': 0, 'other_key': 1, 'hello': 1}
```

However, the following fails:

```python
assert 'HELLO' in d  # << Raise error (Key not normalised)
```

This behavior is inconsistent with `collections.abc.Mapping` (where `__getitem__` is called when `x in my_dict`) so was very surprising to me.

The reason is that `UserDict` overwrite `__contains__` (By removing `del collections.UserDict.__contains__`, the above code works as expected).

From the comment, it looks like this is to support `__missing__`. However when `__missing__` isn't present, the current behavior is surprising.

https://github.com/python/cpython/blob/9f8b9a3506fefeba01a18eda0e06c037393be4bf/Lib/collections/__init__.py#L1118

**Comment by rhettinger** [comment]
We have to leave this as-is. This code has been stable for two decades — changing it now will almost certainly break existing code.  Also, the behavior is consistent with a regular dict and with the notion that the methods are a pass-through to an internal dictionary.

The ABC for Mapping shows show the method could be implemented but it is not a requirement (as demonstrated by *dict* also having its contains method not depend on getitem and by other mappings that rely on missing being called).
